scylla-rust-wrapper/src/collection.rs

```
@@ -57,18 +75,61 @@ pub unsafe extern "C" fn cass_collection_new(
 ) -> *mut CassCollection {
     let capacity = match collection_type {
         // Maps consist of a key and a value, so twice
-        // the number of CqlValue will be stored.
+        // the number of CassCqlValue will be stored.
         CassCollectionType::CASS_COLLECTION_TYPE_MAP => item_count * 2,
         _ => item_count,
     } as usize;
 
     Box::into_raw(Box::new(CassCollection {
         collection_type,
+        data_type: None,
+        capacity,
+        items: Vec::with_capacity(capacity),
+    }))
+}
+
+#[no_mangle]
+unsafe extern "C" fn cass_collection_new_from_data_type(
+    data_type: *const CassDataType,
+    item_count: size_t,
+) -> *mut CassCollection {
+    let data_type = clone_arced(data_type);
+    let (capacity, collection_type) = match data_type.as_ref() {
+        CassDataType::List { .. } => (item_count, CassCollectionType::CASS_COLLECTION_TYPE_LIST),
+        CassDataType::Set { .. } => (item_count, CassCollectionType::CASS_COLLECTION_TYPE_SET),
+        // Maps consist of a key and a value, so twice
+        // the number of CassCqlValue will be stored.
+        CassDataType::Map { .. } => (item_count * 2, CassCollectionType::CASS_COLLECTION_TYPE_MAP),
+        _ => return std::ptr::null_mut(),
```

Does cpp driver silently return a NULL pointer on bad data type passed? If so, that's sad...